src/porepy/models/momentum_balance.py

```diff
@@ -822,12 +844,79 @@ def contact_mechanics_numerical_constant(
         # The scaling factor should not be too large, otherwise the contact problem
         # may be discretized wrongly. I therefore introduce a safety factor here; its
         # value is somewhat arbitrary.
-        softening_factor = 1e-1
+        softening_factor = self.solid.contact_mechanics_scaling()
 
         val = softening_factor * shear_modulus / characteristic_distance
 
         return pp.ad.Scalar(val, name="Contact_mechanics_numerical_constant")
 
+    def contact_mechanics_open_state_characteristic(
+        self, subdomains: list[pp.Grid]
+    ) -> pp.ad.Operator:
+        r"""Characteristic function used in the tangential contact mechanics relation.
+        Can be interpreted as an indicator of the fracture cells in the open state.
+        Used to smoothen the transition between the fracture states and improve
```

Comment:
This is wrong. It's used to make the problem well-posed in the case b_p is zero.